src/query.rs

```diff
@@ -1246,6 +1246,7 @@ add_lints!(
     enum_tuple_variant_field_missing,
     enum_tuple_variant_field_now_doc_hidden,
     enum_unit_variant_changed_kind,
+    enum_unmarked_non_exhaustive,
```

Comment:
Would you mind renaming this lint to follow the implicit convention for items that have stopped being _<something>_? For example, consider `function_abi_no_longer_unwind`.